src/discussions/threads/thread-filter-bar/ThreadFilterBar.jsx

```diff
@@ -0,0 +1,86 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { buildIntlSelectionList } from '../../utils';
+import { SelectableDropdown } from '../../../components/selectable-dropdown';
+import {
+  MyThreadsFilter,
+  AllThreadsFilter,
+  ThreadsStatusFilter,
+  ThreadOrdering,
+} from '../../../data/constants';
+import messages from './messages';
+
+function ThreadFilterBar({ filterSelfThreads, intl }) {
+  let threadsFilterOptions = buildIntlSelectionList(AllThreadsFilter, intl, messages);
+  let defaultThreadFilter = AllThreadsFilter.ALL_THREADS;
+  if (filterSelfThreads) {
+    threadsFilterOptions = buildIntlSelectionList(MyThreadsFilter, intl, messages);
+    defaultThreadFilter = MyThreadsFilter.MY_THREADS;
+  }
+  const [currentThreadFilter, setThreadFilter] = useState(defaultThreadFilter);
+
+  useEffect(() => {
+    setThreadFilter(defaultThreadFilter);
+  }, [filterSelfThreads]);
+
+  const threadsStatusFilterOptions = buildIntlSelectionList(ThreadsStatusFilter, intl, messages);
+  const threadOrderingOptions = buildIntlSelectionList(ThreadOrdering, intl, messages);
+
+  const [currentThreadStatus, setThreadStatus] = useState(ThreadsStatusFilter.ALL);
+  const [currentThreadOrder, setThreadOrder] = useState(ThreadOrdering.BY_LAST_ACTIVITY);
```

Comment:
This should be a dumb component that doesn't store any state. What is the point of storing the current filter internally in the component when it needs to affect the data fetched and the whole UI?

Reply:
@xitij2000

Initially when I wrote this component, I fully expected that it would need a rework to properly store the state so that it can be used by API calls. I was not, and still am not really sure how to approach refactoring this component such that it uses the state appropriately. This is the component that sets the values for filtering/ordering of posts, so it needs to have some mechanism to set those values, whether that be as local state, or in the redux state. I can definitely make it such that it retrieves the existing filtering/ordering from the state, but in order for this component to work properly, I would also have to implement some amount of `slices` or `thunks` code to set/modify that state.

I can try some things out and see what I come up with, but (as you've pointed out) this PR is already somewhat bloated and I feel like this would likely add more bloat. Do you want me to make an attempt at implementation for this, or save it for a future PR?

Reply:
> it needs to have some mechanism to set those values, whether that be as local state, or in the redux state.

I think it only makes sense to store it in redux since storing it locally makes it inaccessible to the rest of the app. So adding code to store it locally doesn't make sense. 

> I can definitely make it such that it retrieves the existing filtering/ordering from the state, but in order for this component to work properly, I would also have to implement some amount of `slices` or `thunks` code to set/modify that state.

The purpose of this task is to implement the UI designs, not to hook up everything properly, As I mentioned earlier there are, and will be other tasks to do that kind of hooking up. 

i.e. I am asking you to remove the existing features, not develop them more thoroughly. You can just plop the component in there as a bit of non-functional UI.

Reply:
@xitij2000

I've removed the state usage from the `PostFilterBar`. It still takes into account whether it should show `My Posts` or `All Posts` (and the associated alternative options), but does not update the filter bar, nor do anything else when you select an option from the dropdown.